### PR TITLE
Add mega linter

### PR DESCRIPTION
**Pull request by shiranr**
Mega linter is an aggregator of linters which can be easily controlled and configured.
If we decide to add/remove components, code, text and such it will be easy to add functionality to the linter accordingly.
This commit also contains fixes to issues the mega linter found such as broken links, table alignments and quotes in the documentation. 
The configuration itself was automatically generated by the mega linter tool and was optimized to our needs.

There is still work which needs to be done over the cspell - spell checker which needs to white list some terms in the documentation

### CODE REVIEW

**Comment by omusavi** [comment]
Great work on this @shiranr! I also did a similar PR but ended up not taking it to completion if you want to use some of the configs from it: #622

**Comment by omusavi** [comment]
> > Great work on this @shiranr! I also did a similar PR but ended up not taking it to completion if you want to use some of the configs from it: #622
> 
> Thank you @omusavi :) I will take a look.
> What was the reason it was abounded, lack of time or an issue with the linter? 

Lack of time and it just dropped off my radar unfortunately

File: .github/workflows/mega-linter.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+# MegaLinter GitHub Action configuration file
```
Comment (shiranr) [inline .github/workflows/mega-linter.yml:2]: Auto generated file.

**Comment by nvuillam** [comment]
@shiranr If I may, MegaLinter v6 has been released and is even better than v5 :)
Just run `npx mega-linter-runner@latest --upgrade` in your branch and see the updates :)

**Comment by shiranr** [comment]
@nvuillam thank you so much for the feedback. 
Will fix it :)

**Comment by nvuillam** [comment]
@shiranr can be solved with either:
- a [`.trivyignore` file](https://aquasecurity.github.io/trivy/v0.21.1/vulnerability/examples/filter/#by-vulnerability-ids) at the root of the repo, containing `DS002`
- adding REPOSITORY_TRIVY in DISABLE_LINTERS or DISABLE_LINTERS_ERRORS list

**Comment by shiranr** [comment]
@nvuillam thanks, I tried to solve it in a more subtle way and not disable the plugin/error. :)

File: .github/workflows/mega-linter.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# More info at https://oxsecurity.github.io/megalinter
+name: MegaLinter
+
+on:
+  # Trigger mega-linter at every push. Action will also be visible from Pull Requests to master
```
Comment (tompaana) [inline .github/workflows/mega-linter.yml:7]:
```suggestion
  # Trigger mega-linter at every push. Action will also be visible from pull requests to main
```

**Comment by nvuillam** [comment]
@shiranr linters can be dumb, they don't know about how and where some code or config is used... that's why I'm not ashamed to often disable what I consider as false positives 😇 

But I have to admit your way is quite elegant.... but has it a real impact or is it just to bypass the linter rule ? ^^
If it is response 2, better disable the rule :)

File: docs/continuous-delivery/secrets-management/README.md [hunk 3]
```diff
@@ -129,8 +129,8 @@ The following steps lay out a clear pathway to creating new secrets and then uti
 
 1. [Install and configure the Databricks CLI](https://docs.databricks.com/user-guide/dev-tools/databricks-cli.html#set-up-the-cli) on your local machine
 2. [Get the Databricks personal access token](https://docs.databricks.com/api/latest/authentication.html#token-management)
-3. [Create a scope for the secrets](https://docs.azuredatabricks.net/user-guide/secrets/secret-scopes.html#create-a-databricks-backed-secret-scope)
-4. [Create secrets](https://docs.azuredatabricks.net/user-guide/secrets/secrets.html)
+3. [Create a scope for the secrets](https://docs.microsoft.com/en-us/azure/databricks/security/secrets/secret-scopes)
```
Comment (tompaana) [inline docs/continuous-delivery/secrets-management/README.md:132]:
```suggestion
3. [Create a scope for the secrets](https://docs.microsoft.com/azure/databricks/security/secrets/secret-scopes)
```

File: docs/continuous-delivery/secrets-management/README.md [hunk 4]
```diff
@@ -129,8 +129,8 @@ The following steps lay out a clear pathway to creating new secrets and then uti
...
+4. [Create secrets](https://docs.microsoft.com/en-us/azure/databricks/security/secrets/)
```
Comment (tompaana) [inline docs/continuous-delivery/secrets-management/README.md:133]:
```suggestion
4. [Create secrets](https://docs.microsoft.com/azure/databricks/security/secrets/)
```

File: docs/agile-development/basics/Ceremonies/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Agile Ceremonies basics
 
-This section has links directing you to best practices for conducting the Agile ceremonies.  After reading through the best practices you should have a basic understanding of the key Agile ceremonies in terms of purpose, value and best practices around conducting and facilating these ceremnonies.
+This section has links directing you to best practices for conducting the Agile ceremonies.  After reading through the best practices you should have a basic understanding of the key Agile ceremonies in terms of purpose, value and best practices around conducting and facilitating these ceremonies.
 
 - [Planning](https://scrumguides.org/scrum-guide.html#sprint-planning 'Sprint Planning')
 - [Refinement](https://docs.microsoft.com/en-us/devops/plan/what-is-agile-development#diligent-backlog-refinement 'Refinement')
```
Comment (tompaana) [inline docs/agile-development/basics/Ceremonies/README.md:6]:
```suggestion
- [Refinement](https://docs.microsoft.com/devops/plan/what-is-agile-development#diligent-backlog-refinement 'Refinement')
```

File: .projector/workItemTemplates/kubernetes-at-the-edge.yml [hunk 6]
```diff
@@ -96,7 +96,7 @@ items:
             acceptanceCriteria:
               - Decision record documenting chosen tool for cluster configuration
             children:
-              - name: Identify tooling to be used to spin up Kubernetes 
+              - name: Identify tooling to be used to spin up Kubernetes
                 type: task
                 description: For baremetal clusters, reference this [decision-tree](https://docs.microsoft.com/en-us/azure/architecture/operator-guides/aks/choose-bare-metal-kubernetes) to determine which tooling shall be used
```
Comment (tompaana) [inline .projector/workItemTemplates/kubernetes-at-the-edge.yml:101]:
```suggestion
                description: For baremetal clusters, reference this [decision-tree](https://docs.microsoft.com/azure/architecture/operator-guides/aks/choose-bare-metal-kubernetes) to determine which tooling shall be used
```

File: docs/observability/pillars/dashboard.md [hunk 7]
```diff
@@ -42,21 +42,21 @@ Here are principles to consider when building dashboards:
...
 - Failure analysis - A report about system failure with details. Workbook template available in gallery.
-- Application Performance Index([Appdex](https://en.wikipedia.org/wiki/Apdex)) - This is a way to measure user satisfaction. It classifies performance into three zones based on a baseline performance threshold T. The template for Appdex is available in Azure Workbooks gallery as well.
+- Application Performance Index([Apdex](https://en.wikipedia.org/wiki/Apdex)) - This is a way to measure user satisfaction. It classifies performance into three zones based on a baseline performance threshold T. The template for Appdex is available in Azure Workbooks gallery as well.
 
-### Application Insights:
+### Application Insights
 
 - [User retention analysis](https://docs.microsoft.com/en-us/azure/azure-monitor/app/usage-retention)
 - [User navigation patterns analysis](https://docs.microsoft.com/en-us/azure/azure-monitor/app/usage-flows)
 - [User session analysis](https://docs.microsoft.com/en-us/azure/azure-monitor/learn/tutorial-users)
```
Comment (tompaana) [inline docs/observability/pillars/dashboard.md:55]:
```suggestion
- [User retention analysis](https://docs.microsoft.com/azure/azure-monitor/app/usage-retention)
- [User navigation patterns analysis](https://docs.microsoft.com/azure/azure-monitor/app/usage-flows)
- [User session analysis](https://docs.microsoft.com/azure/azure-monitor/learn/tutorial-users)
```

**Review by tompaana — APPROVED** [review]
Mega PR for mega linter :)

**Comment by nvuillam** [comment]
It seems we finally have some green :D 
I see that markdown-link-check is expected , but as it is embedded in MegaLinter maybe it can be removed to avoid to make twice the same checks ^^

**Comment by shiranr** [comment]
@nvuillam The lint and link check is part of the old pipeline.
We need to redefine the merge policy to main once this PR is fully approved :).

**Review by nyouens — APPROVED** [review]

**Comment by nvuillam** [comment]
🥳

**Pull request merged**